base/arraymath.jl

```diff
@@ -84,14 +70,14 @@ end
 for f in (:.+, :.-, :.*, :./, :.\, :.^, :.÷, :.%, :.<<, :.>>, :div, :mod, :rem, :&, :|, :$)
     @eval begin
         function ($f){T}(A::Number, B::AbstractArray{T})
-            F = similar(B, promote_array_type($f,typeof(A),typeof(B)))
+            F = similar(B, promote_op($f, typeof(A), T))
             for (iF, iB) in zip(eachindex(F), eachindex(B))
                 @inbounds F[iF] = ($f)(A, B[iB])
             end
             return F
         end
         function ($f){T}(A::AbstractArray{T}, B::Number)
-            F = similar(A, promote_array_type($f,typeof(A),typeof(B)))
+            F = similar(A, promote_op($f, T, typeof(B)))
```

Comment:
`promote_array_type` is responsible for e.g. `typeof(Float32[1,2,3]*1.0) == Vector{Float32}`, even though `Base.promote_op(*, Float32, typeof(1.0)) == Float64` and `typeof(Float32(1)*1.0) == Float64`. If you look at above machinery closer, you'll note that the result of `promote_op` is only used in a few cases, and that's for a reason.